climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return self._impact[haz_ind].aai_agg
+
+    def haz_summary_str(self, run_datetime=None):
+        """ provide a summary string for the hazard part of the forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        str
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return (self.hazard[haz_ind].tag.haz_type +
+                '_' +
+                self.haz_model +
+                '_run' +
+                run_datetime.strftime('%Y%m%d%H') +
+                '_event' +
+                self.event_date.strftime('%Y%m%d')
+                )
+
+    def summary_str(self, run_datetime=None):
+        """ provide a summary string for the impact forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        str
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        return (self.haz_summary_str(run_datetime) +
+                '_' +
+                self.exposure_name
+                )
+
+    def lead_time(self, run_datetime=None):
+        """ provide the lead time for the impact forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            select the used hazard by the run_datetime.
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        return self.event_date - run_datetime
+
+    def calc(self, force_reassign=False, check_plot=False):
+        """ calculate the impact using
+        exposure, hazard, and vulnerabilty.
+
+        Parameters
+        ----------
+        force_reassign: bool, optional
+            Reassign centroids.
+        check_plot: bool, optional
+            Show plot of expected annual impact if True.
+        """
+        # generate folders
+        if not Path(FORECAST_PLOT_DIR).exists():
+            Path(FORECAST_PLOT_DIR).mkdir(parents=True)
+
+        # force reassign
+        if force_reassign:
+            self.exposure.assign_centroids(self.hazard)
+        # calc impact
+        for ind_i, haz_i in enumerate(self.hazard):
+            self._impact[ind_i].calc(self.exposure,
+                         self.vulnerability,
+                         haz_i,
+                         save_mat=True)
+            if check_plot:
+                self._impact[ind_i].plot_hexbin_eai_exposure()
+
+    def plot_imp_map(self,
+                     run_datetime=None,
+                     save_fig=True,
+                     close_fig=True,
```

Comment:
Why would you want to have such an option as `close_fig`? You can always close the fig by 
```
plot_imp_map()
plt.close()
```
I think you should remove this everywhere.

Reply:
First of all. I of course changed the default to False. 
Second: I introduced it, because this function might be running everyday (as it does on my ETH server) and I don't the plots to show up, but to be closed directly after saving. Does this necessarily have to be deleted from the code (and the tutorials) I do not think it bothers anyone if it stays there...

Reply:
This is all perfectly valid. 

Just to be clear, here are my reasons for advocating for a removal are:
1 - logic: a plotting function shouldn't not return the plot
2 - necessity: there is absolutely no need for such an option. This functionality is 100% redundant.
3- CLIMADA consistency: no other plotting function has this option.
If you use that functionality all the time, is it much work to add `plt.close()` to your code instead of `close_fig=True` ?